lib/rucio/core/replica.py

```diff
@@ -1022,19 +1068,29 @@ def _list_replicas(dataset_clause, file_clause, state_clause, show_pfns,
                                 # does it match with the client? if not, it's an outgoing connection
                                 # therefore the internal proxy must be prepended
                                 if client_location['site'] != replica_site:
-                                    root_proxy_internal = config_get('root-proxy-internal',    # section
-                                                                     client_location['site'],  # option
-                                                                     default='',               # empty string to circumvent exception
-                                                                     session=session)
-
-                                    if root_proxy_internal:
-                                        # TODO: XCache does not seem to grab signed URLs. Doublecheck with XCache devs.
-                                        #       For now -> skip prepending XCache for GCS.
-                                        if 'storage.googleapis.com' in pfn or 'atlas-google-cloud.cern.ch' in pfn or 'amazonaws.com' in pfn:
-                                            pass  # ATLAS HACK
-                                        else:
-                                            # don't forget to mangle gfal-style davs URL into generic https URL
-                                            pfn = 'root://' + root_proxy_internal + '//' + pfn.replace('davs://', 'https://')
+                                    cache_site = config_get('clientcachemap', client_location['site'], default='', session=session)
+                                    if cache_site != '':
+                                        # print('client', client_location['site'], 'has cache:', cache_site)
+                                        # print('filename', name)
+                                        selected_prefix = get_multi_cache_prefix(cache_site, name)
+                                        if selected_prefix:
+                                            pfn = 'root://' + selected_prefix + '//' + pfn.replace('davs://', 'root://')
+                                    else:
+                                        # print('site:', client_location['site'], 'has no cache')
```

Comment:
remove all prints, and potentially replace with comments

Reply:
I took added one more test assert that makes sure no cache sites are not prefixed.